File: ruby/arrays/assignment_and_overwriting.rb
#!/usr/bin/ruby2.7

array = []

puts("Not initialised, before assignments: array = []\n\n")
puts(array.inspect, "\n")

array = [0, 0, 0]

puts("Initialised, before assignments: array = [0, 0, 0]\n\n")
puts(array.inspect, "\n")

array[1] = 6

puts("After assignment: array[1] = 6\n\n")
puts(array.inspect, "\n")

array[1] = 10

puts("After overwriting: array[1] = 10\n\n")
puts(array.inspect, "\n")

File: javascript/arrays/assignment_and_overwriting.js
#!/usr/bin/env node

{
  let array = [];

  console.log("Not initialised, before assignments: let array = []\n");
  console.log(array); console.log();
}

{
  let array = [0, 0, 0];

  console.log("Initialised, before assignments: let array = [0, 0, 0]\n");
  console.log(array); console.log();

  array[1] = 6;

  console.log("After assignment: array[1] = 6\n");
  console.log(array); console.log();

  array[1] = 10;

  console.log("After overwriting: array[1] = 10\n");
  console.log(array); console.log();
}
